packages/data-grid/src/internal/gridHooks.tsx

```diff
@@ -185,18 +185,26 @@ export function useVisibleRowRange(
       return NumberRange.empty;
     }
     const firstRowHeight = rowHeight + 1; // First row has an extra 1px
+
     const start =
       scrollTop > firstRowHeight
         ? 1 + Math.floor((scrollTop - firstRowHeight) / rowHeight)
         : 0;
     let endPos = scrollTop + clientMidHeight;
+
     if (start === 0) {
       endPos -= 1;
     }
     const end = Math.min(
       rowCount,
       Math.max(start, Math.ceil(endPos / rowHeight))
     );
+
+    // Scroll Top not returning to 0 after pagination. Guard to ensure issues with ScrollTop throw error
+    if (start > end) {
```

Comment:
wondering if it should be >=? also will need cs :)

Reply:
NumberRange doesn't throw an error when when start = end so i think either is fine
cs = cypress test ?

Reply:
a changeset

Reply:
added a changeset - doesn't seem to have been picked up by the github action. Am i doing something wrong ?

Reply:
I think it automatically updates when you add one, not sure why it hasn't, do you know @joshwooding ? could just be taking time

Reply:
Not sure, it should just work automatically